packages/text-bitmap/src/BitmapFont.ts

```
@@ -226,4 +236,293 @@ export class BitmapFont
+                baseTexture = new BaseTexture(canvas, { resolution });
+            }
+
+            // Measure glyph dimensions
+            const metrics = TextMetrics.measureText(chars[i], style, false, canvas);
+            const width = metrics.width;
+            const height = metrics.height;
+
+            maxCharHeight = Math.max(height + metrics.fontProperties.descent, maxCharHeight);
+
+            // Wrap line once full row has been rendered
+            if ((width * resolution) + positionX >= lineWidth)
+            {
+                --i;
+                positionY += maxCharHeight * resolution;
+                positionX = 0;
+                maxCharHeight = 0;
+                continue;
+            }
+
+            drawGlyph(// eslint-disable-line @typescript-eslint/no-use-before-define
+                canvas,
+                context,
+                metrics,
+                positionX,
+                positionY,
+                resolution,
+                style
+            );
+
+            // Unique (numeric) ID mapping to this glyph
+            const id = metrics.text.charCodeAt(0);
+
+            // Create a texture holding just the glyph
+            textures[id] = new Texture(baseTexture, new Rectangle(positionX / resolution, positionY / resolution,
+                width,
+                height));
+
+            fontData.char[id] = {
+                id,
+                page: id,
+                x: 0,
+                y: 0,
+                width,
+                height,
+                xoffset: 0,
+                yoffset: 0,
+                xadvance: width - (style.dropShadow ? style.dropShadowDistance : 0),
+            };
+            fontData.page[id] = {
+                id,
+                file: '',
+            };
+
+            positionX += (width + (2 * BMT_PADDING)) * resolution;
+        }
+
+        const font = new BitmapFont(fontData, textures);
+
+        if (name)
```

What do you think should be the behavior trying to set a name that's already in use? We could throw an error or automatically destroy the existing font? There's a potential memory leak here if we don't destroy all the textures and someone uses BitmapFont.from over and over again to overwrite.

---

I could add a throws clause to prevent name collision (and a test for that). Do you want another API for unregistering these fonts?

---

We already have `BitmapFont.uninstall`. I think a throw that lets users know they should uninstall sounds good. And yes, a test for that.

---

@bigtimebuddy I added a test for name collision (that checks if an error is thrown).